src/main/java/seedu/address/model/person/Person.java

```diff
@@ -17,41 +22,67 @@ public class Person {
 
     // Identity fields
     private final Name name;
-    private final Phone phone;
-    private final Email email;
-
-    // Data fields
-    private final Address address;
+    private Email email = null;
+    private Phone phone = null;
+    private Slack slack = null;
+    private Telegram telegram = null;
     private final Set<Tag> tags = new HashSet<>();
 
     /**
      * Every field must be present and not null.
      */
-    public Person(Name name, Phone phone, Email email, Address address, Set<Tag> tags) {
-        requireAllNonNull(name, phone, email, address, tags);
+    public Person(Name name) {
+        requireAllNonNull(name);
         this.name = name;
-        this.phone = phone;
-        this.email = email;
-        this.address = address;
-        this.tags.addAll(tags);
     }
 
     public Name getName() {
         return name;
     }
 
+    public Email getEmail() {
+        return email;
+    }
+    public Person setEmail(Email email) {
+        this.email = email;
+        return this;
+    }
+
     public Phone getPhone() {
         return phone;
     }
+    public Person setPhone(Phone phone) {
+        this.phone = phone;
+        return this;
+    }
 
-    public Email getEmail() {
-        return email;
+    public Slack getSlack() {
+        return slack;
+    }
+    public Person setSlack(Slack slack) {
+        this.slack = slack;
+        return this;
+    }
+
+    public Telegram getTelegram() {
+        return telegram;
+    }
+    public Person setTelegram(Telegram telegram) {
+        this.telegram = telegram;
+        return this;
```

Comment:
As we discussed before, we should use a hashMap to store this contact information such as `HashMap<ContactEnum, Contact>`